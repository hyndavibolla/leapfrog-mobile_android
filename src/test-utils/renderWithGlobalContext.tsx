import React, { ReactElement } from 'react';
import { render } from '@testing-library/react-native';

import { getInitialState, GlobalProvider } from '../state-mgmt/GlobalState';
import { Deps, IGlobalState, IReducer } from '../models/general';
import { getMockDeps } from './getMockDeps';
import { getMockReducer } from './getMockReducer';

export const renderWithGlobalContext = (
  children: ReactElement,
  deps: Deps = getMockDeps(),
  initState: IGlobalState = getInitialState(),
  combinedReducers: IReducer<IGlobalState, any> = getMockReducer()
) => {
  const renderAPI = render(
    <GlobalProvider deps={deps} initState={initState} combinedReducers={combinedReducers}>
      {children}
    </GlobalProvider>
  );
  return { ...renderAPI, mockReducer: combinedReducers, deps };
};
