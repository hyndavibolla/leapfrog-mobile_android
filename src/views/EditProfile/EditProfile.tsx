import React, { memo, useMemo, useContext, useEffect } from 'react';

import { ENV } from '_constants';
import { WebView } from '_components/WebView';
import { ConnectionBanner } from '_views/shared/ConnectionBanner';
import { useTestingHelper } from '_utils/useTestingHelper';
import { useAsyncCallback } from '_utils/useAsyncCallback';
import { ITokenSet } from '_models/general';
import { GlobalContext } from '_state_mgmt/GlobalState';

export interface Props {
  route: { params: { uri: string } };
}

export const EditProfile = ({ route }: Props) => {
  const { getTestIdProps } = useTestingHelper('edit-profile');

  const { deps } = useContext(GlobalContext);
  const { MAX_LOAD_ATTEMPTS: maxLoadAttempts, ATTEMPT_DELAY_MS: attemptDelay } = ENV.WEBVIEWS;
  const [getTokenSet, isGettingToken = true, , tokenSet] = useAsyncCallback<any, ITokenSet>(async () => {
    return await deps.apiService.getTokenSetAsync();
  }, []);
  const uri = route?.params?.uri;
  const retry = useMemo(() => ({ attempts: maxLoadAttempts, delayMs: attemptDelay }), [attemptDelay, maxLoadAttempts]);

  useEffect(() => {
    getTokenSet();
  }, [getTokenSet]);

  return (
    <>
      <ConnectionBanner />
      {isGettingToken ? null : (
        <WebView
          {...getTestIdProps('webview')}
          shouldShowLoader={true}
          source={{ uri, headers: { 'onetrust-mask': 'true', accessToken: tokenSet?.accessToken } }}
          retry={retry}
        />
      )}
    </>
  );
};

export default memo(EditProfile);
