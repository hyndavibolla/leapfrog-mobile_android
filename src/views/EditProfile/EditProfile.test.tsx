import React from 'react';
import { act } from 'react-test-renderer';

import { renderWithGlobalContext } from '_test_utils/renderWithGlobalContext';
import { EditProfile, Props } from './EditProfile';
import { wait } from '_utils/wait';

describe('EditProfile', () => {
  let props: Props;

  beforeEach(() => {
    props = {
      route: { params: { uri: 'http://www.shopyourway.com' } }
    };
  });

  it('should render', async () => {
    const { toJSON } = renderWithGlobalContext(<EditProfile {...props} />);
    await act(() => wait(0));
    expect(toJSON()).toMatchSnapshot();
  });
});
