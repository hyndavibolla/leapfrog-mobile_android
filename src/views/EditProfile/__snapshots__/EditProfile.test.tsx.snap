// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProfile should render 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <RNCWebView
    accessibilityLabel="edit-profile-webview"
    accessible={true}
    cacheEnabled={true}
    injectedJavaScript="true;"
    injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
    injectedJavaScriptForMainFrameOnly={true}
    javaScriptEnabled={true}
    messagingEnabled={true}
    onContentProcessDidTerminate={[Function]}
    onError={[Function]}
    onHttpError={[Function]}
    onLoad={[Function]}
    onLoadingError={[Function]}
    onLoadingFinish={[Function]}
    onLoadingProgress={[Function]}
    onLoadingStart={[Function]}
    onMessage={[Function]}
    onShouldStartLoadWithRequest={[Function]}
    setSupportMultipleWindows={false}
    sharedCookiesEnabled={true}
    source={
      Object {
        "headers": Object {
          "accessToken": undefined,
          "onetrust-mask": "true",
        },
        "uri": "http://www.shopyourway.com",
      }
    }
    startInLoadingState={true}
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        undefined,
      ]
    }
    testID="edit-profile-webview"
    textInteractionEnabled={true}
    thirdPartyCookiesEnabled={true}
    useSharedProcessPool={true}
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5F7FA",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "height": "100%",
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
  >
    <div>
      Spinner
    </div>
  </View>
</View>
`;
